packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -509,15 +509,20 @@ class DevFS {
           await generator.recompile(mainPath, invalidatedFiles,
               outputPath:  dillOutputPath ?? fs.path.join(getBuildDirectory(), 'app.dill'),
               packagesFilePath : _packagesFilePath);
-      final String compiledBinary = compilerOutput?.outputFilename;
-      if (compiledBinary != null && compiledBinary.isNotEmpty) {
-        final Uri entryUri = fs.path.toUri(projectRootPath != null ?
-            fs.path.relative(pathToReload, from: projectRootPath):
-            pathToReload);
-        if (!dirtyEntries.containsKey(entryUri)) {
-          final DevFSFileContent content = new DevFSFileContent(fs.file(compiledBinary));
-          dirtyEntries[entryUri] = content;
-          numBytes += content.size;
+      // Don't send full kernel file that would overwrite what VM already
+      // started loading from.
+      if (!bundleFirstUpload) {
+        final String compiledBinary = compilerOutput?.outputFilename;
+        if (compiledBinary != null && compiledBinary.isNotEmpty) {
+          final Uri entryUri = fs.path.toUri(projectRootPath != null ?
+          fs.path.relative(pathToReload, from: projectRootPath) :
+          pathToReload);
```

Comment:
nit: indentation here is confusing

Reply:
Yeah, IJ auto-indenter took over as soon as added curly braces.